### PR TITLE
ref: aktualizr client + secondary interface

### PR DESCRIPTION
**Pull request by kbushgit**
Signed-off-by: Kostiantyn Bushko <kostiantyn.bushko@yahoo.com>

### CODE REVIEW

File: src/libaktualizr/primary/aktualizr.cc [hunk 1]
```diff
@@ -1,10 +1,11 @@
-#include "aktualizr.h"
-
-#include <chrono>
-
 #include <sodium.h>
+#include <chrono>
```
Comment (patrickvacek) [inline src/libaktualizr/primary/aktualizr.cc:2]: Not a big deal, but C++ headers should be listed before external libs.

File: docs/ota-client-guide/modules/ROOT/pages/libaktualizr-getstarted.adoc [hunk 2]
```diff
@@ -131,9 +131,9 @@ Libaktualizr provides a C++ API for fetching information about available updates
 
 If you're yet not familiar with OTA Connect concepts such as "campaigns", have a look at the https://connect.ota.here.com/#/campaigns[OTA web app] first. You might need to register for a free developer account first.
 
-The main library header is https://github.com/advancedtelematic/aktualizr/blob/master/src/libaktualizr/primary/aktualizr.h[`primary/aktualizr.h`]. It also includes few other libaktualizr headers.
+The main library header is https://github.com/advancedtelematic/aktualizr/blob/master/include/libaktualizr/aktualizr.h[`libaktualizr/aktualizr.h`]. It also includes few other libaktualizr headers.
 
-To use the API, add the `src/libaktualizr` directory to your include path and add `# include "primary/aktualizr.h"` to your source file.
+To use the API, add the `aktualizr/include` directory to your include path and add `# include <libaktualizr/aktualizr.h>` to your source file.
```
Comment (patrickvacek) [inline docs/ota-client-guide/modules/ROOT/pages/libaktualizr-getstarted.adoc:136]: Not your change, but why is there a space between the `#` and `include`?

File: src/libaktualizr/primary/aktualizr.cc (hunk 1 above)
Comment (kbushgit) [reply src/libaktualizr/primary/aktualizr.cc:2]: It's not my fault,  ```make format``` makes this changed(:

File: include/libaktualizr/aktualizr.h [hunk 3]
```diff
@@ -8,10 +8,15 @@
...
-#include "utilities/apiqueue.h"
+#include "libaktualizr/secondary/secondaryinterface.h"
+
+class SotaUptaneClient;
+class INvStorage;
+class SecondaryInfo;
+
+namespace api {
+class CommandQueue;
+}
```
[the remark this answers is not in the document]
Comment (kbushgit) [reply include/libaktualizr/aktualizr.h:19]: I tried it but the compiler generate an error like:
```
forward declaration of class cannot have a nested name specifier
```

File: src/libaktualizr/primary/aktualizr.cc (hunk 1 above)
Comment (patrickvacek) [reply src/libaktualizr/primary/aktualizr.cc:2]: Even if you put empty lines between the C++ and external header sections?

File: include/libaktualizr/aktualizr.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (patrickvacek) [reply include/libaktualizr/aktualizr.h:19]: Okay, no worries.

File: src/libaktualizr/primary/aktualizr.cc (hunk 1 above)
Comment (kbushgit) [reply src/libaktualizr/primary/aktualizr.cc:2]: thanks it works

**Comment by patrickvacek** [comment]
I'm not so sure about putting some headers in a `secondary` subdirectory of the `include/libaktualizr` directory. Those headers are still specifically for the Primary to use, so to me it seems like a confusing separation. Or was there a specific goal you were trying to achieve with that?

**Comment by kbushgit** [comment]
> I'm not so sure about putting some headers in a `secondary` subdirectory of the `include/libaktualizr` directory. Those headers are still specifically for the Primary to use, so to me it seems like a confusing separation. Or was there a specific goal you were trying to achieve with that?

No, there is no specific purpose, I'm just not sure if we have to mimic the structure of folders in the public headers as it is in the ```src```? 
Is the idea, to put all those headers in one folder more comfortable for you?

File: src/virtual_secondary/partialverificationsecondary.cc [hunk 4]
```diff
@@ -37,14 +37,15 @@ data::InstallationResult PartialVerificationSecondary::putMetadata(const Target
   }
 
   // TODO(OTA-2484): check for expiration and version downgrade
-  root_ = Uptane::Root(RepositoryType::Director(), Utils::parseJSON(director_root), root_);
+  Uptane::Root root(Root::Policy::kAcceptAll);
+  root = Uptane::Root(RepositoryType::Director(), Utils::parseJSON(director_root), root);
```
Comment (patrickvacek) [inline src/virtual_secondary/partialverificationsecondary.cc:41]: This is a downgrade in functionality by not storing the Root to check it the next time around. However, this whole class is woefully incomplete, so it doesn't really matter. It needs a lot of work and that isn't part of the scope here.

**Pull request merged**